src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,243 @@
+        .div(kElevatorGearing);
+  }
+
+  // height = C * rot * 2/75 -> rot = height * 75/2C
+  public static Angle heightToRotations(Distance height) {
+    return height
+        .times(kElevatorGearing)
+        .div(kElevatorDrumCircumference.times(kCarageFactor))
+        .times(Rotations.one());
+  }
+
+  public static AngularVelocity heightToRotations(LinearVelocity vel) {
+    // rot/s = (G)/(C*2/vel)
+    return Rotations.one()
+        .times(kElevatorGearing)
+        .div((kElevatorDrumCircumference.times(kCarageFactor)).div(vel));
+  }
+
+  public Distance getElevatorHeight() {
+    return rotationsToHeight(elevatorMotor1.getPosition().getValue());
+  }
+
+  public Command goUp(Supplier<Double> upPower) {
+    return runEnd(
+        () -> {
+          elevatorMotor1.set(Math.abs(upPower.get()) * elevatorOutput);
+        },
+        () -> elevatorMotor1.set(0));
+  }
+
+  public Command goDown(Supplier<Double> downPower) {
+    return runEnd(
+        () -> {
+          elevatorMotor1.set(Math.abs(downPower.get()) * -elevatorOutput);
+        },
+        () -> elevatorMotor1.set(0));
+  }
+
+  private void disableSoftLimits() {
+    elevatorMotor1
+        .getConfigurator()
+        .apply(
+            new SoftwareLimitSwitchConfigs()
+                .withForwardSoftLimitEnable(false)
+                .withReverseSoftLimitEnable(false));
+  }
+
+  private void enableSoftLimits() {
+    elevatorMotor1.getConfigurator().apply(elvSoftLimit);
+  }
+
+  public Command limitHit() {
+    return runOnce(this::disableSoftLimits)
+        .andThen(goDown(() -> 1.0).until(elvLimitSwitch::get))
+        .andThen(zeroMotorEncoder())
+        .andThen(runOnce(this::enableSoftLimits));
+  }
+
+  public Command zeroMotorEncoder() {
+    return runOnce(
+        () -> {
+          elevatorMotor1.setPosition(0);
+        });
+  }
+
+  public Command changeElevation(Distance heightLevel) {
+    return runOnce(
+        () -> {
+          Angle adjustedSetpoint = heightToRotations(heightLevel);
+          elevatorMotor1.setControl(new MotionMagicVoltage(adjustedSetpoint));
+          SmartDashboard.putNumber("Elevator/Setpoint (Inches)", heightLevel.in(Inches));
+          SmartDashboard.putNumber("Elevator/Setpoint Rotations", adjustedSetpoint.in(Rotations));
+        });
+  }
+
+  public Command L0() {
+    return changeElevation(Constants.ElevatorConstants.kL0Height);
+  }
+
+  public Command L1() {
+    return changeElevation(Constants.ElevatorConstants.kL1Height);
+  }
+
+  public Command L2() {
+    return changeElevation(Constants.ElevatorConstants.kL2Height);
+  }
+
+  public Command L3() {
+    return changeElevation(Constants.ElevatorConstants.kL3Height);
+  }
+
+  public Command L4() {
+    return changeElevation(Constants.ElevatorConstants.kL4Height);
+  }
+
+  @Override
+  public void periodic() {
+    SmartDashboard.putBoolean("Elevator limit switch state", elvLimitSwitch.get());
```

Comment:
The / organizes the values in AdvantageScope

```suggestion
    SmartDashboard.putBoolean("Elevator/limit switch state", elvLimitSwitch.get());
```